pennylane/_queuing.py

```diff
@@ -93,13 +104,16 @@ def remove(cls, obj):
         Args:
             obj: the object to be removed
         """
+        # TODO: this method should remove only from `cls.active_context`, *not*
+        # all active contexts. However this will require a refactor in
+        # the template decorator and the operation recorder.
         for context in cls._active_contexts:
             # We use the duck-typing approach to assume that the underlying remove
             # behaves like list.remove and throws a ValueError if the operator
             # is not in the list
             try:
                 context._remove(obj)  # pylint: disable=protected-access
-            except ValueError:
+            except (ValueError, KeyError):
```

Comment:
Is it worth adding to the comment above why a `KeyError` might be raised (since the `ValueError` is already explained)?

Reply:
Updated!